tests/ocp/sriov/README.md

```diff
@@ -0,0 +1,557 @@
+
+## Test Case Structure
+
+### Ginkgo Test Patterns
+
+#### Basic Test Structure
+
+```go
+var _ = Describe(
+    "TestFeatureName", 
+    Ordered, 
+    Label(tsparams.LabelFeatureName), 
+    ContinueOnFailure, 
+    func() {
+        var (
+            // Shared test variables
+        )
+
+        BeforeAll(func() {
+            // Setup required for all tests in this Describe block
+            By("Setting up test environment")
+            // ... setup code ...
+        })
+
+        AfterAll(func() {
+            // Cleanup after all tests
+            By("Cleaning up test resources")
+            // ... cleanup code ...
+        })
+
+        BeforeEach(func() {
+            // Setup for each test
+        })
+
+        AfterEach(func() {
+            // Cleanup after each test
+        })
+
+        It("should perform specific test action", reportxml.ID("12345"), func() {
+            By("Step description")
+            // Test implementation
+        })
+    })
+```
+
+#### Test Organization Principles
+
+1. **Ordered Tests**: Use `Ordered` when tests must run in sequence
+2. **Labels**: Always use labels for test filtering (`Label(tsparams.LabelFeatureName)`)
+3. **ContinueOnFailure**: Use when appropriate to gather more diagnostic information
+4. **Descriptive Names**: Test descriptions should clearly state what is being tested
+
+### Test Labels
+
+#### Label Naming Convention
+
+- Use lowercase with hyphens or underscores
+- Be specific and descriptive
+- Group related tests under the same label
+- Define labels in `internal/tsparams/consts.go`
+
+Example:
+```go
+const (
+    LabelSuite = "sriov"
+    LabelFeatureName = "feature-name"
+    LabelHWEnabled = "sriov-hw-enabled"
+)
+```
+
+### Test IDs
+
+- Use `reportxml.ID("12345")` for each test case
+- Test IDs should be unique and traceable to test case management systems
+- Include test ID in test description or comments when helpful
+
+## Internal Packages
+
+### Test Parameters Package (`tsparams`)
+
+Purpose: Centralize constants, variables, and configuration for the test suite.
+
+#### Structure
+
+```go
+package tsparams
+
+const (
+    // Labels
+    LabelSuite = "sriov"
+    LabelFeatureName = "feature-name"
+    
+    // Namespaces
+    TestNamespaceName = "sriov-tests"
+    
+    // Timeouts
+    WaitTimeout = 3 * time.Minute
+    DefaultTimeout = 300 * time.Second
+)
+
+var (
+    // Labels list for test selection
+    Labels = []string{LabelSuite, LabelFeatureName}
+    
+    // Reporter configuration
+    ReporterCRDsToDump = []k8sreporter.CRData{...}
+    ReporterNamespacesToDump = map[string]string{...}
+)
+```
+
+#### Best Practices
+
+1. **Constants**: Use for values that don't change (labels, namespaces, default values)
+2. **Variables**: Use for values that may be configured or computed
+3. **Timeouts**: Define reasonable defaults; make them configurable when needed
+4. **Labels**: Aggregate in a `Labels` variable for suite-level filtering
+
```

Comment:
We also need to revert (at a minimum at test suite level) if any config change is made. For example, we can have a list of CRs that we modify for the whole test suites, and save a snapshot at BeforeSuite, then reapply all these CRs at AfterSuite (or at BeforeAll and AfterAll), assuming we may not always want to revert at test case level, which can significantly increase the execution time especially if node reboot is involved. Although it shouldn't prevent revert at test case level when needed.